retrieval/discovery/marathon/url.go

```diff
@@ -0,0 +1,14 @@
+package marathon
+
+import (
+	"fmt"
+	"math/rand"
+)
+
+const appsURLFormat string = "http://%s/v2/apps/?embed=apps.tasks"
+
+// RandomAppsURL randomly selects a server from an array and creates an URL pointing to the app list
+func RandomAppsURL(servers []string) string {
```

Comment:
I don't see this as an issue as the list of Marathon masters does not change on a regular basis, but is an infrequent configuration change. Adding/removing slave nodes is much more common, but is also no issue here as the slaves do not need to be known in advance.